src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimerViewModel.cs

```diff
@@ -0,0 +1,209 @@
+            }
+        }
+
+        private void SetRunningTimeEntry(Toggl.TogglTimeEntryView item)
+        {
+            ResetRunningTimeEntry(true);
+            runningTimeEntry = item;
+            TimeEntryLabelViewModel = item.ToTimeEntryLabelViewModel();
+            DurationText = Toggl.FormatDurationInSecondsHHMMSS(item.DurationInSeconds);
+            DurationPanelToolTip = "started at " + item.StartTimeString;
+        }
+
+        private void ResetRunningTimeEntry(bool running, bool forceUpdate = false)
+        {
+            var changedState = IsRunning != running;
+
+            if (!(changedState || forceUpdate || this.acceptNextUpdate))
+                return;
+
+            acceptNextUpdate = false;
+
+            IsRunning = running;
+            Description = "";
+            DurationText = "";
+        }
+
+        private void onStoppedTimerState()
+        {
+            using (Performance.Measure("timer responding to OnStoppedTimerState"))
+            {
+                secondsTimer.IsEnabled = false;
+                ResetRunningTimeEntry(false);
+                runningTimeEntry = default(Toggl.TogglTimeEntryView);
+            }
+        }
+
+        #region controlling
+
+        public void startStop()
+        {
+            this.acceptNextUpdate = true;
+
+            if (IsRunning)
+            {
+                this.stop();
+            }
+            else
+            {
+                this.start();
+            }
+        }
+
+        public void tryOpenEditViewIfRunning(string focusedField)
+        {
+            if (IsRunning)
+            {
+                using (Performance.Measure("opening edit view from timer, focussing " + focusedField))
+                {
+                    Toggl.Edit(this.runningTimeEntry.GUID, false, focusedField);
+                }
+            }
+        }
+
+        private void start()
+        {
+            using (Performance.Measure("starting time entry from timer"))
+            {
+                var guid = Toggl.Start(
+                    Description,
+                    "",
+                    completedProject.TaskID,
+                    completedProject.ProjectID,
+                    "",
+                    completedProject.Tags,
+                    isMiniTimer
+                    );
+
+                if (completedProject.Billable)
+                {
+                    Toggl.SetTimeEntryBillable(guid, true);
+                }
+
+                this.clearSelectedProject();
+            }
+        }
+
+        private void stop()
+        {
+            using (Performance.Measure("stopping time entry from timer"))
+            {
+                Toggl.Stop(isMiniTimer);
+            }
+        }
+
+        #endregion
+
+        public void clearSelectedProject()
+        {
+            ProjectLabelViewModel = null;
+            completedProject = default;
+        }
+
+        public void DescriptionAutoCompleteConfirm(object e)

```

Comment:
the argument can be strongly typed here

Reply:
Thank you for such a nice and detailed feedback! I agree with your comments, and I see, there is a couple of things I lost out of my view (about ```StartStopClick```, and I really didn't look much in the MiniTimer's code).